libsolidity/ast/AST.h

```diff
@@ -1521,7 +1530,7 @@ class WhileStatement: public BreakableStatement
 /**
  * For loop statement
  */
-class ForStatement: public BreakableStatement, public Scopable
+class ForStatement: public BreakableStatement, public Scopable, public ScopeOpener
```

Comment:
And what is the difference between Scopable and ScopeOpener?

Reply:
Scopable stores the scope it is defined in and scope opener defines a new scope.